File: conformance/cases/Presence/presence.go
package presence

// The three presence/nullability combinations expressible across all
// extractor languages (SPEC §7).
type Presence struct {
	RequiredNonNull  string  `json:"requiredNonNull"`
	RequiredNullable *string `json:"requiredNullable"`          // pointer: present, may be null
	OptionalNonNull  string  `json:"optionalNonNull,omitempty"` // may be absent (note: Go drops "" too)
}

File: conformance/cases/Presence/Presence.java
package conformance;

import javax.annotation.Nonnull;
import java.util.Optional;

/** The three presence/nullability combinations expressible in both languages (SPEC §7). */
public class Presence {
    @Nonnull public String requiredNonNull;
    public String requiredNullable;             // plain reference: present, may be null
    public Optional<String> optionalNonNull;    // may be absent, never null
}
